src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -229,6 +227,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 	get onRequestAddInstanceToGroup(): Event<IRequestAddInstanceToGroupEvent> { return this._onRequestAddInstanceToGroup.event; }
 
 	constructor(
+		private readonly _messageShown: boolean,
```

Comment:
This would be easier by doing `let migrationMessageShown = false;`  at the top of the file, in this area:

https://github.com/microsoft/vscode/blob/3d2309d4e746802df6ae42f4f4ae9bcde6ce03c0/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L64-L91

Then terminal service doesn't need to know about it